docs/SPEC.md

```diff
@@ -839,6 +833,29 @@ Side effects can occur in two ways.
 
 A function produces side effects when it is explicitly declared to have side effects or when it calls a function that itself produces side effects.
 
+## Package initialization
+
+A package with no imports is initialized by first initializing and assigning all declared options and then subsequently initializing and assigning all declared variables.
+In particular all package options are initialized before any variables.
+Options cannot depend on any variables declared in the same package block.
+If a package has imports, all imported packages are initialized before any options or variables.
+
+### Variable initialization
+
+Variables are initialized and assigned the same value regardless of the order they are declared in the source.
+A variable cannot have a direct or indirect dependency on itself.
+
+### Option initialization
+
+An option cannot have dependencies on any other options assigned in the same package block.
```

Comment:
Does this mean that options can depend on options from other packages?

Reply:
Good question. The way the spec is worded, yes. Allowing options to have dependencies on external options certainly won't introduce any nondeterminism so it's possible. It doesn't affect the implementation at all, but it could introduce some unexpected behavior for the user that's difficult to debug. I'll defer to you on whether we should keep it.